js/packages/web/src/components/Analytics/index.tsx

```diff
@@ -80,35 +80,37 @@ export function AnalyticsProvider(props: { children: React.ReactNode }) {
 
   function pageview(path: string) {
     if (!gtag) return;
-    gtag('event', 'page_view', {
-      page_location: window.location.href, // important to overwrite to keep fragments
-      page_path: path, // React router provides the # route as a regular path
+    track('page_view', {
+      path,
     });
   }
 
   function track(
     action: string,
-    attributes?: {
+    attributes: {
       category?: string;
       label?: string;
       value?: number;
       sol_value?: number;
       [key: string]: string | number | undefined;
-    } & Partial<CustomEventDimensions>,
+    } & Partial<CustomEventDimensions> = {},
   ) {
     if (!gtag) return;
+    const { category, label, sol_value, value, ...otherAttributes } =
+      attributes;
     gtag('event', action, {
-      event_category: attributes?.category,
-      event_label: attributes?.label,
-      ...(attributes?.sol_value && solPrice
+      event_category: category,
+      event_label: label,
+      page_location: window.location.href,
+      ...(sol_value && solPrice
         ? {
-            value: attributes.sol_value * solPrice, //Google Analytics likes this one in USD :)
-            sol_value: attributes.sol_value,
+            value: sol_value * solPrice, //Google Analytics likes this one in USD :)
+            sol_value: sol_value,
           }
```

Comment:
These fields seem like good ones for additional attributes. Not something on the track function itself.

Reply:
I figured since the sol_value is a very valuable and common attribute, and that "value" is directly linked to it it made sense to just standardise it in the function.

But sure, we can generalise it :)